### PR TITLE
Track names of moderated packages to disallow as new package names

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: app/lib/package/backend.dart [hunk 1]
```diff
@@ -938,7 +938,7 @@ Future<_ValidatedUpload> _parseAndValidateUpload(
   final pubspec = Pubspec.fromYaml(archive.pubspecContent);
   if (!await nameTracker.accept(pubspec.name)) {
     throw GenericProcessingException(
-        'Package name is too similar to another package.');
+        'Package name is too similar to another active or moderated package.');
```
Comment (isoos) [inline app/lib/package/backend.dart]: not sure: moderated -> reserved ?

File: app/lib/package/name_tracker.dart [hunk 2]
```diff
@@ -43,18 +44,26 @@ class NameTracker {
...
+      _reservedNames.contains(name);
 
   /// Whether to accept the upload attempt of a given package [name].
   ///
   /// Either the package [name] should exists, or it should be different enough
-  /// from already existing package names. An example for the rejection:
-  /// `long_name` will be rejected, if package `longname` or `lon_gname` exists.
+  /// from already existing active or moderated package names. An example for
+  /// the rejection: `long_name` will be rejected, if package `longname` or
+  /// `lon_gname` exists.
```
Comment (isoos) [inline app/lib/package/name_tracker.dart]: nit: `lon_gname` -> `long_name`

Comment (szakarias) [reply app/lib/package/name_tracker.dart]: Isn't it supposed to be `lon_gname`, or am I misunderstanding the comment?

Comment (isoos) [reply app/lib/package/name_tracker.dart]: My bad, automatic spell check error on my part...

File: app/lib/package/backend.dart (hunk 1 above)
Comment (szakarias) [reply app/lib/package/backend.dart]: I want to keep `moderated` as this is the term we use when we remove the package. But maybe the internal use of `reserved` is a bit confusing. I'll add a bit of documentation.

Comment (sigurdm) [reply app/lib/package/backend.dart]: This is an error that will show up on `pub publish` right?

As an end-user I might get confused by the term 'moderated' without further context.

**Pull request merged**